### PR TITLE
#1648 Add more tests for ZSink combinators that concatenate leftovers

### PR DESCRIPTION
**Pull request by msosnicki**

### CODE REVIEW

File: streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala [hunk 1]
```diff
@@ -1109,6 +1127,15 @@ class SinkSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRunt
     unsafeRun(sinkIteration(sink, 1).map(_ must_=== ((None, Chunk.empty))))
   }
 
+  private def untilLeftoverAppend = {
```
Comment (iravid) [inline streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala]:
```suggestion
  private def untilOutputLeftoverAppend = {
```

Comment (msosnicki) [reply streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala]: Thanks @iravid If there are any other combinators that append leftovers and are not covered here, please let me know

**Comment by jdegoes** [comment]
@msosnicki Can you resolve conflicts so we can take another look at this? Thank you!

**Comment by jdegoes** [comment]
@msosnicki I love more tests! This PR has been open for a while, but please re-open if your schedule frees up enough to push this one through. Thanks for your work on it!

**Pull request closed without merging**